presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveClient.java

```diff
@@ -36,5 +49,63 @@ public void initialize(String host, int port, String databaseName, String timeZo
         }
 
         setup(host, port, databaseName, timeZone);
+
+        checkArgument(hiveVersionMajor > 0, "Invalid hiveVersionMajor: %s", hiveVersionMajor);
+        this.hiveVersionMajor = hiveVersionMajor;
+    }
+
+    protected int getHiveVersionMajor()
+    {
+        checkState(hiveVersionMajor > 0, "hiveVersionMajor not set");
+        return hiveVersionMajor;
+    }
+
+    @Override
+    public void testGetPartitionSplitsTableOfflinePartition()
+    {
+        if (getHiveVersionMajor() >= 2) {
+            throw new SkipException("ALTER TABLE .. ENABLE OFFLINE was removed in Hive 2.0 and this is a prerequisite for this test");
+        }
+
+        super.testGetPartitionSplitsTableOfflinePartition();
+    }
+
+    @Override
+    public void testTypesRcBinary()
+            throws Exception
+    {
+        if (getHiveVersionMajor() >= 3) {
+            // TODO (https://github.com/prestosql/presto/issues/1218) requires https://issues.apache.org/jira/browse/HIVE-22167
+            assertThatThrownBy(super::testTypesRcBinary)
+                    .isInstanceOf(AssertionError.class)
+                    .hasMessage("expected [2011-05-06 01:23:09.123] but found [2011-05-06 07:08:09.123]");
+            return;
+        }
+        super.testTypesRcBinary();
+    }
+
+    @Override
+    public void testTypesParquet()
+            throws Exception
+    {
+        if (getHiveVersionMajor() >= 3) {
+            // TODO (https://github.com/prestosql/presto/issues/1218) requires https://issues.apache.org/jira/browse/HIVE-21002
+            assertThatThrownBy(super::testTypesParquet)
+                    .isInstanceOf(AssertionError.class)
+                    .hasMessage("expected [2011-05-06 01:23:09.123] but found [2011-05-06 07:08:09.123]");
+            return;
+        }
+        super.testTypesParquet();
+    }
+
+    @Override
+    public void testMismatchSchemaTable()
```

Comment:
What is the nature of this failure that is making us exclude it?

Reply:
The varchar to integer coercion fails in Hive >=3.